staging/src/k8s.io/apiserver/pkg/server/options/etcd.go

```diff
@@ -226,7 +226,7 @@ func (s *EtcdOptions) ApplyWithStorageFactoryTo(factory serverstorage.StorageFac
 }
 
 func (s *EtcdOptions) addEtcdHealthEndpoint(c *server.Config) error {
-	healthCheck, err := storagefactory.CreateHealthCheck(s.StorageConfig, c.StopNotify())
```

Comment:
this doesn't look related to this PR, is this a separate fix?

Reply:
it is related - I wanted to have a single signal and for resource quota I want to stop it only after all requests are drained (which btw seems to be fine also in this case).

[But I'm happy to move it to a separate PR too.]